### PR TITLE
add yelp_review_full

### PR DESCRIPTION
**Pull request by hfawaz**
This corresponds to the Yelp-5 requested in https://github.com/huggingface/datasets/issues/353
I included the dataset card.

### CODE REVIEW

File: datasets/yelp_review_full/yelp_review_full.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    def _info(self):
+        features = datasets.Features(
+            {
+                "label": datasets.features.ClassLabel(
+                    names=[
+                        "0 star",
+                        "1 star",
+                        "2 stars",
+                        "3 stars",
+                        "4 stars",
```
Comment (lhoestq) [inline datasets/yelp_review_full/yelp_review_full.py:83]: aren't the stars between 1 and 5 actually ?

Can you update the labels and regenerate the dataset_infos.json file please ?

**Review by lhoestq — APPROVED** [review]
Nice thank you :) 

Just one last nit

File: datasets/yelp_review_full/yelp_review_full.py (hunk 1 above)
Comment (lhoestq) [inline datasets/yelp_review_full/yelp_review_full.py:83]:
```suggestion
                        "1 star",
                        "2 star",
                        "3 stars",
                        "4 stars",
                        "5 stars",
```

File: datasets/yelp_review_full/dataset_infos.json [hunk 2]
```diff
@@ -0,0 +1 @@
...s={649--657},\n  year={2015}\n}\n", "homepage": "https://www.yelp.com/dataset", "license": "https://s3-media3.fl.yelpcdn.com/assets/srv0/engineering_pages/bea5c1e92bf3/assets/vendor/yelp-dataset-agreement.pdf", "features": {"label": {"num_classes": 5, "names": ["0 star", "1 star", "2 stars", "3 stars", "4 stars"], "names_file": null, "id": null, "_type": "ClassLabel"}, "text": {"dtype": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "yelp_review_full", "config_name": "yelp_review_full", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 483811814, "num_examples": 650000, "dataset_name": "yelp_review_full"}, "test": {"name": "test", "num_bytes": 37271208, "num_examples": 50000, "dataset_name": "yelp_review_full"}}, "download_checksums": {"https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbZlU4dXhHTFhZQU0": {"num_bytes": 196146693, "checksum": "9f4dd0a449885e1b5679cf79cd03f06f157190e53f4af4a325aa7bcc9381bee7"}}, "download_size": 196146693, "post_processing_size": null, "dataset_size": 521083022, "size_in_bytes": 717229715}}
```
Comment (lhoestq) [inline datasets/yelp_review_full/dataset_infos.json:1]:
```suggestion
{"yelp_review_full": {"description": "The Yelp reviews dataset consists of reviews from Yelp. It is extracted from the Yelp Dataset Challenge 2015 data.\nThe Yelp reviews full star dataset is constructed by Xiang Zhang (xiang.zhang@nyu.edu) from the above dataset.\nIt is first used as a text classification benchmark in the following paper: Xiang Zhang, Junbo Zhao, Yann LeCun.\nCharacter-level Convolutional Networks for Text Classification. Advances in Neural Information Processing Systems 28 (NIPS 2015).\n", "citation": "@inproceedings{zhang2015character,\n  title={Character-level convolutional networks for text classification},\n  author={Zhang, Xiang and Zhao, Junbo and LeCun, Yann},\n  booktitle={Advances in neural information processing systems},\n  pages={649--657},\n  year={2015}\n}\n", "homepage": "https://www.yelp.com/dataset", "license": "https://s3-media3.fl.yelpcdn.com/assets/srv0/engineering_pages/bea5c1e92bf3/assets/vendor/yelp-dataset-agreement.pdf", "features": {"label": {"num_classes": 5, "names": ["1 star", "2 star", "3 stars", "4 stars", "5 stars"], "names_file": null, "id": null, "_type": "ClassLabel"}, "text": {"dtype": "string", "id": null, "_type": "Value"}}, "post_processed": null, "supervised_keys": null, "builder_name": "yelp_review_full", "config_name": "yelp_review_full", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 483811814, "num_examples": 650000, "dataset_name": "yelp_review_full"}, "test": {"name": "test", "num_bytes": 37271208, "num_examples": 50000, "dataset_name": "yelp_review_full"}}, "download_checksums": {"https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbZlU4dXhHTFhZQU0": {"num_bytes": 196146693, "checksum": "9f4dd0a449885e1b5679cf79cd03f06f157190e53f4af4a325aa7bcc9381bee7"}}, "download_size": 196146693, "post_processing_size": null, "dataset_size": 521083022, "size_in_bytes": 717229715}}
```

File: datasets/yelp_review_full/yelp_review_full.py (hunk 1 above)
Comment (lhoestq) [reply datasets/yelp_review_full/yelp_review_full.py]: actually I just did the change, all good now !

Comment (hfawaz) [reply datasets/yelp_review_full/yelp_review_full.py]: Thanks for the review!

**Pull request merged**